### DISCUSSION TITLE
Clarification on the /spawnrate command

### DISCUSSION THREAD

**Discussion by hazylua** [Pryaxis/TShock#1953] [Q&A] [answered]
Just a quick question about how the /spawnrate command works: is it just a modifier on top of the current world spawn rate or does it modify the current spawn rate directly?
I'm assuming it's a modifier because when you use the command at night in the forest it shows that the spawn rate is currently at 600,  which is the base rate from daytime forest , when it' supposed to be 360~ or something like that.
Sorry if the question sounds dumb, just curious on how the command works and how it affects my worlds.

**Comment by hakusaro** [comment, marked as answer]
Somewhat confusingly, the lower the spawnrate, the more things will spawn. TShock modifies `NPC.defaultSpawnRate`. Higher difficulty modes will multiply this value by a multiplier less than one, which reduces the "spawn rate" value but causes more things to spawn. The way that the game actually processes this is a bit complex, but suffice it to say that if you want a lot of things, set it lower. If you want a lot less, set it higher. Default is 600, and higher difficulties act as a multiplier on top. E.g., https://github.com/Pryaxis/Sources/blob/b66ed5b2/1.4.0.4/Terraria/NPC.cs#L73326

**Reply by hakusaro** [reply]
```csharp
						spawnRate = defaultSpawnRate;
						maxSpawns = defaultMaxSpawns;
						if (Main.hardMode)
						{
							spawnRate = (int)((double)defaultSpawnRate * 0.9);
							maxSpawns = defaultMaxSpawns + 1;
						}
```
